libs/math/include/math/tensor.hpp

```diff
@@ -1223,19 +1223,31 @@ Tensor<T, C> &Tensor<T, C>::operator=(TensorSlice const &slice)
 template <typename T, typename C>
 bool Tensor<T, C>::Resize(SizeVector const &shape, bool copy)
 {
-  Tensor old_tensor = *this;
+  // if the shape is exactly the same and a copy of value is required, dont do anything
+  if ((this->shape() == shape) && copy)
+  {
+    return true;
+  }
+
+  // this line does a shallow copy for speedy initializion of a tensor
```

Comment:
pedantic point:
don't say "this line does a " in a comment - we know you're commenting about this line